### PR TITLE
GH-1691A: As a smith I want to improve "add missing imports" functionality in code completion and quick fixes

### PR DESCRIPTION
**Pull request by mmews-n4**
See #1691.

### CODE REVIEW

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/contentassist/CompletionWithImportsTest.xtend [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+						export class A2 {}'''];
+	}
+
+	@Test
+	def void test01() {
+		test('''
+			let x = new A1<|>
+		''', ''' 
+			(A1, Class, MA, , , 00000, , , , ([0:12 - 0:14], A1), [([0:0 - 0:0], import {A1} from "MA";
+			)], [], , )
```
Comment (mor-n4) [inline tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/contentassist/CompletionWithImportsTest.xtend]: What about applying the changes sent by the server and then using the resulting changed code for comparison (to make the test more readable)?
If `TextEdit`s are sent, here, then there is already method `AbstractIdeTest#applyTextEdits()` that might help.

**Comment by mor-n4** [comment]
Two things I noticed (but they are more a problem of content assist in general, not the "add missing import"):

```javascript
import {A02} from "Temp";

A02
A01/* here, content assist does not work (caused by missing ; in line above) */
```

```javascript
import * as N from "Temp";
N.A01
N./* here, content assist offers only A01, not A02, A03, etc. */
N.A/* here it works as expected */
```
(module `Temp` exports classes `A01` ... `A04`)

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/contentassist/CompletionWithImportsTest.xtend (hunk 1 above)
Comment (mmews-n4) [reply tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/contentassist/CompletionWithImportsTest.xtend]: Thanks, good idea, but I also want to validate the label, icon, description besides the actual replacement string.

**Comment by mmews-n4** [comment]
Regarding the "two things": I noticed that too, but think it is out of scope. I created a new ticket, however.

**Comment by mmews-n4** [comment]
Thanks! Builds green, merging part A.

**Pull request merged**